src/main/resources/lep/custom/xm/xm/entity/queue/system/AcceptSystemEvent$$INDEX_RELOAD$$around.groovy

```diff
@@ -0,0 +1,101 @@
+package lep.custom.xm.xm.entity.queue.system
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.icthh.xm.commons.domainevent.domain.DomainEvent
+import com.icthh.xm.commons.domainevent.domain.DomainEventPayload
+import com.icthh.xm.commons.domainevent.domain.enums.DefaultDomainEventOperation
+import com.icthh.xm.commons.domainevent.domain.enums.DefaultDomainEventSource
+import com.icthh.xm.commons.tenant.TenantContext
+import com.icthh.xm.commons.tenant.TenantContextUtils
+import com.icthh.xm.commons.topic.service.KafkaTemplateService
+import com.icthh.xm.ms.entity.domain.XmEntity
+import com.icthh.xm.ms.entity.repository.XmEntityRepository
+import groovy.transform.Field
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.PageRequest
+import org.springframework.data.domain.Pageable
+import org.springframework.data.jpa.domain.Specification
+
+final String APP_NAME = "entity"
+final String DB_TOPIC_FORMAT = "event.%s.db"
+final int PAGE_SIZE = 1000;
+
+@Field ObjectMapper objectMapper = lepContext.services.objectMapper
+
+Logger log = LoggerFactory.getLogger(getClass())
+Map<String, Object> eventData = lepContext.inArgs.event?.data as Map<String, Object>
+TenantContext tenantContext = lepContext.tenantContext
+KafkaTemplateService kafkaTemplateService = lepContext.templates.kafka
+XmEntityRepository xmEntityRepository = lepContext.repositories.xmEntity
+
+String tenantKey = TenantContextUtils.getRequiredTenantKeyValue(tenantContext)
+Specification<XmEntity> spec = buildSpecification(eventData)
+
+long count = xmEntityRepository.count(spec)
```

Comment:
why it supports only XmEntity? Can we design approach for fetching any entity by type defined in the command?